media/js/src/GridAsset.jsx

```diff
@@ -100,7 +101,15 @@ export default class GridAsset extends React.Component {
         const type = this.asset.getType();
         const a = this.props.asset.annotations[annotation];
 
-        this.setState({selectedAnnotation: a});
+        this.setState({selectedAnnotation: a},
+            function(){
+                if (this.state.selectedAnnotation){
+                    this.setState({annotationLabel: `annotation: ${this.state.selectedAnnotation.title}`});
+                } else {
+                    this.setState({annotationLabel: `asset: ${this.props.asset.title}`});
+                }
```

Comment:
There are a few problems with this logic here.

First, the nested `this` on line 106 refers to the immediately surrounding function being defined at line 105. To get to the components `this`, you would make a new variable to refer to it, e.g. `const me = this;` in the outer context.

Second, we don't want to use `setState` in a chain like this - `setState` is meant to be used in a single pass, and calling it multiple times like this can have unintended side effects.

The logic here can be refactored for a single setState call, e.g. something like:
```
let label = `asset: ${this.props.asset.title}`;
if (a) {
    label = `annotation: ${a.title}`;
}

this.setState({
    selectedAnnotation: a,
    label: label
});
```

I renamed `annotationLabel` to just `label`, since it may refer to either the asset or the annotation.